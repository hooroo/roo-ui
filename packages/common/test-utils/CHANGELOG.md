# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

<a name="0.4.0"></a>
# [0.4.0](https://github.com/hooroo/roo-ui/compare/v0.3.3...v0.4.0) (2018-04-26)


### Features

* add theming ([bee896a](https://github.com/hooroo/roo-ui/commit/bee896a))
