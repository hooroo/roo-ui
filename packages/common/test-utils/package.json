{
  "name": "@roo-ui/test-utils",
  "version": "0.0.0",
  "main": "dist/index.js",
  "publishConfig": {
    "access": "public"
  },
  "scripts": {
    "build": "babel src -d dist --copy-files"
  }
}
