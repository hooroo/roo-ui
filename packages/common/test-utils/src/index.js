export { default as shallowWithTheme } from './shallowWithTheme'; // eslint-disable-line import/prefer-default-export
