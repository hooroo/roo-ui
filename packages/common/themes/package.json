{
  "name": "@roo-ui/themes",
  "version": "0.4.0",
  "main": "dist/index.js",
  "publishConfig": {
    "access": "public"
  },
  "scripts": {
    "build": "babel src -d dist --copy-files"
  }
}
