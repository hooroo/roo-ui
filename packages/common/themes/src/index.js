import qantas from './qantas';

export { qantas }; // eslint-disable-line import/prefer-default-export
