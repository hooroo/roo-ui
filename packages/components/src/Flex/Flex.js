import styled from 'styled-components';
import tag from 'clean-tag';
import {
  space,
  width,
  height,
  color,
  alignItems,
  justifyContent,
  flexWrap,
  flexDirection,
} from 'styled-system';

const Flex = styled(tag)`
  display: flex;

  ${space}
  ${width}
  ${height}
  ${color}
  ${alignItems}
  ${justifyContent}
  ${flexWrap}
  ${flexDirection};
`;

Flex.propTypes = {
  ...space.propTypes,
  ...width.propTypes,
  ...height.propTypes,
  ...color.propTypes,
  ...alignItems.propTypes,
  ...justifyContent.propTypes,
  ...flexWrap.propTypes,
  ...flexDirection.propTypes,
};

Flex.displayName = 'Flex';

export default Flex;
