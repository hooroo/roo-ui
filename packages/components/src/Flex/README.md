# Flex

<!-- STORY -->

## Installation

```shell
$ yarn add @roo-ui/components
```

## Example

```js
import { Flex, Box } from '@roo-ui/components';

export default (
  <Flex>
    <Box>Flex</Box>
    <Box>Box</Box>
  </Flex>
);
```

## Customization

This component can be customized with [styled-system](https://jxnblk.com/styled-system) by passing props for `alignContent`, `alignItems`, `justifyContent`, `flexWrap`, `flexFlow` & `flexDirection` [functions](http://jxnblk.com/styled-system/table)
as well as props supported by [`<Box />`](../Box/README.md)
