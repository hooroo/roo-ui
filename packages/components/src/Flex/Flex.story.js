import React from 'react';
import { storiesOf } from '@storybook/react';
import { withDocs } from 'storybook-readme';

import Flex from './Flex';
import { Box } from '..';
import README from './README.md';

storiesOf('Components|Flex', module)
  .addDecorator(withDocs(README))
  .add('default', () => (
    <Flex>
      <Box width={1 / 2} p={3} m={3} bg="grey.6">Flex</Box>
      <Box width={1 / 2} p={3} m={3} bg="grey.6">Box</Box>
    </Flex>
  ));
