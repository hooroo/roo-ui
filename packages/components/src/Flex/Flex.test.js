import React from 'react';
import { qantas as theme } from '@roo-ui/themes';
import { shallowWithTheme } from '@roo-ui/test-utils';
import { axe } from 'jest-axe';

import Flex from '.';

describe('<Flex />', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowWithTheme(<Flex>Hello world</Flex>, theme);
  });

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('has no accessibility errors', async () => {
    expect(await axe(wrapper.html())).toHaveNoViolations();
  });
});
