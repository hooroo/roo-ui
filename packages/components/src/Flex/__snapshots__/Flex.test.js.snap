// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Flex /> renders correctly 1`] = `
<StyledComponent
  forwardedRef={null}
>
  Hello world
</StyledComponent>
`;
